src/app/helpers/tableTemplate/pagedTableTemplate.ts

```diff
@@ -13,11 +13,211 @@ import {
+
+    this.filterEvent$
+      .pipe(
+        debounceTime(defaultDebounceTime),
+        distinctUntilChanged(),
+        takeUntil(this.unsubscribe)
+      )
+      .subscribe(
+        () => this.getPageData(),
+        // Filter event doesn't have an error output
+        (err) => {}
+      );
+  }
+
+  public ngOnInit() {
+    this.createColumns();
+
+    if (this.route) {
+      const models = retrieveResolvers(this.route.snapshot.data as PageInfo);
+      if (!models) {
+        this.failure = true;
+        return;
+      }
+      this.models = models;
+    }
+
+    this.getPageData();
+  }
+
+  /**
+   * Set the new page number
+   * @param pageInfo Information about current page
+   */
+  public setPage(pageInfo: TablePage) {
+    this.pageNumber = pageInfo.offset;
+    this.filters.paging = {
+      page: pageInfo.offset + 1,
+    };
+
+    this.getPageData();
+  }
+
+  /**
+   * Handle filter events
+   * @param filter Filter text
+   */
+  public onFilter(filter: string) {
+    if (!filter) {
+      this.filters.filter = undefined;
+    } else {
+      this.filters.filter = {
+        // TODO Figure out how to get this typing working
+        [this.filterKey as any]: { contains: filter },
+      };
+    }
+
+    this.filterEvent$.next(filter);
+  }
+
+  /**
+   * Handle sorting events
+   * @param event Sort Event
+   */
+  public onSort(event: SortEvent) {
+    if (!event.newValue) {
+      this.filters.sorting = undefined;
+    } else {
+      this.filters.sorting = {
+        orderBy: this.sortKeys[event.column.prop] as string,
+        direction: event.newValue,
+      };
+    }
+
+    this.getPageData();
+  }
+
+  /**
+   * Retrieve page data for the table. If you are trying to customize
+   * the request, override `apiAction` instead
+   */
+  public getPageData() {
+    this.loadingData = true;
+    this.rows = [];
+
+    this.apiAction(this.filters, this.getUrlParameters(this))
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(
+        (models) => {
+          this.rows = this.createRows(models);
+          this.loadingData = false;
+
+          if (models.length > 0) {
+            const meta = models[0].getMetadata();
+            this.totalModels = meta.paging.total;
+          } else {
+            this.totalModels = 0;
+          }
+        },
+        (err: ApiErrorDetails) => {
+          this.error = err;
+          this.rows = undefined;
+          this.loadingData = false;
+        }
+      );
+  }
+
+  /**
+   * API Request. Override this if you wish to make a more
+   * customized request
+   */
+  protected apiAction(
+    filters: Filters<M>,
+    args: AbstractModel[]
+  ): Observable<M[]> {
+    return this.api.filter(filters, ...args);
+  }
+
+  /**
+   * Create datatable rows
+   */
+  protected createRows(models: M[]) {
+    return models.map((model) => {
+      const row = {};
+      this.columnData.forEach(
+        (column) => (row[column.key] = column.transform(model))
```

Comment:
`TableColumn` has a `pipe` property that accepts a `pipeTransform` that could replace this